x-pack/plugins/security/server/session_management/session.ts

```diff
@@ -150,7 +151,7 @@ export class Session {
   async get(request: KibanaRequest) {
     const sessionCookieValue = await this.options.sessionCookie.get(request);
     if (!sessionCookieValue) {
-      return null;
+      return new SessionMissingError();
```

Comment:
using my alternative approach, this line would look similar to:

```ts
      return { error: new SessionMissingError(), value: null };
```

(rinse and repeat for the other error conditions below)